### PR TITLE
refactor: shared ScrollAutomation

### PR DESCRIPTION
**Pull request by LavrovArtem**

### CODE REVIEW

File: test/functional/fixtures/api/es-next/take-screenshot/pages/nested-iframe.html [hunk 1]
```diff
@@ -7,9 +7,10 @@
 <iframe width="640" height="480" src="same-domain-iframe.html">
 </iframe>
 <script>
-    const testCafeAutomation = window['%testCafeAutomation%'];
-    const ScrollAutomation   = testCafeAutomation.Scroll;
-    const origRunScroll      = ScrollAutomation.prototype.run;
+    debugger;
```
Comment (miherlosev) [inline test/functional/fixtures/api/es-next/take-screenshot/pages/nested-iframe.html:10]: Remove this `debugger` keyword.

File: src/client/core/utils/shared/values/axis-values.ts [hunk 2]
```diff
@@ -44,4 +44,11 @@ export default class AxisValues<T> implements AxisValuesData<T> {
 
         return this;
     }
+
+    public round (this: AxisValues<number>, fn = Math.round): AxisValues<number> {
```
Comment (miherlosev) [inline src/client/core/utils/shared/values/axis-values.ts:48]: `this` is a bad name for the function argument. Rename here and in others places in this file.

`public round (axisValues: AxisValues<number>, roundFn = Math.round): AxisValues<number>`

**Review by miherlosev — CHANGES_REQUESTED** [review]

File: src/client/core/utils/shared/values/axis-values.ts (hunk 2 above)
Comment (LavrovArtem) [reply src/client/core/utils/shared/values/axis-values.ts:48]: `this` is not a function parameter. This is the type of `this` context. We cannot apply this function to `AxisValues<boolean>`. Therefore, we define the type of `this`.

Comment (miherlosev) [reply src/client/core/utils/shared/values/axis-values.ts:48]: Ok

**Review by miherlosev — APPROVED** [review]

**Review by AndreyBelym — APPROVED** [review]

**Pull request merged**